platform.go

```diff
@@ -64,16 +66,19 @@ func (e *PlatformHandlerError) Unwrap() error {
 // PlatformKeyData represents the data exchanged between this package and
 // platform implementations via the PlatformKeyDataHandler.
 type PlatformKeyData struct {
-	EncodedHandle    []byte // The JSON encoded platform handle
-	EncryptedPayload []byte // The encrypted payload
+	Version       int
+	EncodedHandle []byte // The JSON encoded platform handle
+	KDFAlg        crypto.Hash
+
+	AuthMode AuthMode
```

Comment:
There's nothing in the tests inside `mockPlatformKeyDataHandler` that checks we pass the expected values for these new fields.

Reply:
I added some validation for the `mockPlatformKeyDataHandler` fields. I found a missing field in the legacy tests because of it. Let me know how it looks. I hardcoded `crypto.SHA256` in `mockProtectKeys` for the legacy keyData test bases as there aren't any test cases that test for different KDF algorithms.

Reply:
Thanks, I added a comment for this separately before responding to this thread: https://github.com/snapcore/secboot/pull/265#discussion_r1450624305